include/LogicBlock/Z3Logic.hpp

```diff
@@ -66,8 +66,8 @@ namespace z3logic {
         void        internalReset() override;
 
     public:
-        Z3LogicBlock(z3::context& ctx, z3::solver& solver, bool convertWhenAssert = true):
-            LogicBlock(convertWhenAssert), Z3Base(ctx), solver(solver) {}
+        Z3LogicBlock(z3::context& context, z3::solver& sol, bool convert = true):
+            LogicBlock(convert), Z3Base(context), solver(sol) {}

```

Comment:
Name changes are a bit inconsistent, why solver to sol, but ctx to context?
```suggestion
        Z3LogicBlock(z3::context& context, z3::solver& solver, bool convert = true):
            LogicBlock(convert), Z3Base(context), solver(solver) {}

```

Reply:
That's because of the inconsistency in the spelling of the class members. As stated above, these name changes are merely to resolve shadowing warnings.
If you really want consistency here, then you'd probably have to change the class member name in some way.